apps/asap-server/src/migrations/1618575365-move-short-text-to-admin-notes.ts

```diff
@@ -0,0 +1,56 @@
+/* istanbul ignore file */
+import { RestResearchOutput, Results, Squidex } from '@asap-hub/squidex';
+import { Migration } from '../handlers/webhooks/webhook-run-migrations';
+
+export default class MoveResearchOutputTextToDescription extends Migration {
+  up = async (): Promise<void> => {
+    const squidexClient = new Squidex<RestResearchOutput>('research-outputs', {
+      unpublished: true,
+    });
+
+    let pointer = 0;
+    let result: Results<RestResearchOutput>;
+
+    do {
+      result = await squidexClient.fetch({
+        $top: 10,
+        $skip: pointer,
+        $orderby: 'created asc',
+      });
+
+      for (const researchOutput of result.items) {
+        await squidexClient.patch(researchOutput.id, {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          adminNotes: (researchOutput.data as any).shortText,
+        });
+      }
+
+      pointer += 10;
+    } while (pointer < result.total);
+  };
+  down = async (): Promise<void> => {
+    const squidexClient = new Squidex<RestResearchOutput>('research-outputs', {
+      unpublished: true,
+    });
+
+    let pointer = 0;
+    let result: Results<RestResearchOutput>;
+
+    do {
+      result = await squidexClient.fetch({
+        $top: 10,
+        $skip: pointer,
+        $orderby: 'created asc',
+      });
+
+      for (const researchOutput of result.items) {
+        await squidexClient.patch(researchOutput.id, {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          adminNotes: null as any, // types dont cover null values
```

Comment:
Because we're not deleting the `shortText` just yet we can use it in the rollback and just keep a failsafe condition of null for once it is gone
`adminNotes: researchOutput.data.shortText || null as any, // types dont cover null values`

Reply:
I guess I need some coffee. Forget about it. 😅